daprdocs/content/en/operations/components/setup-bindings/supported-bindings/mqtt.md

```diff
@@ -21,20 +21,97 @@ spec:
   version: v1
   metadata:
   - name: url
-    value: mqtt[s]://[username][:password]@host.domain[:port]
+    value: "tcp://[username][:password]@host.domain[:port]"
```

Comment:
@Hugome - is this a breaking change? does the component now expect a different URL string?

Reply:
In the old component the scheme was ignored and forced to `tcp`
https://github.com/dapr/components-contrib/blob/a43a0d908bd8f32e0ea4b24607fd2ab2c4902ec5/bindings/mqtt/mqtt.go#L127

To ensure backward compatibly we can add a little `if scheme == mqtt => scheme = tcp` and `if scheme == mqtts => scheme = ssl` ?
(Same discussion with @pkedy on https://github.com/dapr/components-contrib/pull/771#discussion_r606337985)

Reply:
@Hugome Please, keep backwards compatibility so people don't need to change their current component to upgrade.

Reply:
@artursouza @Hugome is this resolved?

Reply:
@AaronCrawfis I just committed some code that takes `mqtt`, `mqtts`, and `tcps` and converts them to the correct schemes that work with the MQTT client and preserve backward compatibility. IMO this PR _should_ update the docs for both the binding and pubsub to prefer `tcp` and `ssl` as the schemes.